### PR TITLE
MV2: all trackerdb patterns + bugfixes

### PR DESCRIPTION
**Pull request by chrmod**

### CODE REVIEW

**Comment by GRadziejewski** [comment]
Uncategorized tracker is visible on Ghostery panel. 
Open website for example: https://politico.com
image

**Comment by GRadziejewski** [comment]
After selecting "Site Trusted" Ad-blocker is still block some trackers.
Open website for example: https://reuters.com/

Screenshot_14 - bug

**Comment by GRadziejewski** [comment]
After selecting "Site Restricted" few trackers are still Allowed. On one side it's not meant to be blocked or not blocked by default, but in this situation it's blocked and it's showing the opposite state. 
Open website for example: https://espn.com/

image

**Comment by GRadziejewski** [comment]
After pausing Ghostery, few trackers are still blocked by Ad-Blocking. 
Open website for example: https://bbc.com/

Screenshot_29 - bug

**Comment by GRadziejewski** [comment]
@chrmod tests are finished for this iteration.

**Comment by chrmod** [comment]
[ghostery-chrome-v8.11.0.zip](https://github.com/ghostery/ghostery-extension/files/11819104/ghostery-chrome-v8.11.0.zip)

@GRadziejewski another version is ready for testing

**Comment by GRadziejewski** [comment]
**Not fixed:**
https://github.com/ghostery/ghostery-extension/pull/1196#issuecomment-1592976799
image

**Fixed**:
https://github.com/ghostery/ghostery-extension/pull/1196#issuecomment-1597476943
image

https://github.com/ghostery/ghostery-extension/pull/1196#issuecomment-1597462967
image
image

https://github.com/ghostery/ghostery-extension/pull/1196#issuecomment-1598376798
image

https://github.com/ghostery/ghostery-extension/pull/1196#issuecomment-1598423102
image

**Comment by GRadziejewski** [comment]
Tests of https://github.com/ghostery/ghostery-extension/pull/1196#issuecomment-1601169321 are finished.

File: extension-manifest-v2/_locales/en/messages.json [hunk 1]
```diff
@@ -1856,6 +1856,12 @@
   "trackers_preview": {
     "message": "Trackers Preview"
   },
+  "tracking_not_detected": {
+    "message": "Tracking not detected"
+  },
+  "trusted": {
+    "message": "Trused"
```
Comment (philipp-classen) [inline extension-manifest-v2/_locales/en/messages.json:1863]:
```suggestion
    "message": "Trusted"
```

File: extension-manifest-v2/app/panel/components/Blocking/Tracker.jsx [hunk 2]
```diff
@@ -270,29 +262,63 @@ class Tracker extends React.Component {
...
+					}
+					// not blocked and website not restricted
+					if (!tracker.blocked && this.props.sitePolicy !== 1) {
+						return (
+							<div className="trk-common-stats-container">
+								{t('allowed')}
+							</div>
+						);
+					}
+					if (tracker.blocked || this.props.sitePolicy === 1) {
```
Comment (philipp-classen) [inline extension-manifest-v2/app/panel/components/Blocking/Tracker.jsx:313]: This is a negation of the previous condition, so it is always true.

File: extension-manifest-v2/app/panel/components/Blocking/Tracker.jsx [hunk 3]
```diff
@@ -270,29 +262,63 @@ class Tracker extends React.Component {
...
+						return (
+							<div className="trk-common-stats-container">
+								{t('tracking_not_detected')}
+							</div>
+						);
+					}
+					return null;
```
Comment (philipp-classen) [inline extension-manifest-v2/app/panel/components/Blocking/Tracker.jsx:320]: This line is unreachable.

File: extension-manifest-v2/src/utils/matcher.js [hunk 4]
```diff
@@ -81,12 +81,19 @@ export function isBug(details) {
 	const { engine } = bugDb;
 
 	const request = fromWebRequestDetails(details);
+	let isFilterBlocker = false;
 
-	const matches = engine.getPatternMetadata(request);
+	let matches = engine.getPatternMetadata(request);
+
+	if (matches.length > 0) {
+		isFilterBlocker = true;
```
Comment (philipp-classen) [inline extension-manifest-v2/src/utils/matcher.js:89]: Maybe there is a better name than "isFilterBlocker". When it is used, it is referred to as "isFilterMatched".

I wonder if it should be rather should like "isInPatternDB"/"existsInPatternDb"/"hasPatternRule"/"foundInPatternMetadata". I'm lacking an intuition what it means to have a non-empty match in the pattern meta.

Or maybe "blockingShouldBeHandledByPatternEngine" / "usePatternEngineForBlocking"?

Comment (chrmod) [reply extension-manifest-v2/src/utils/matcher.js:89]: all good points. I'm also not sure how to name it best, but my hope for this code is to be short lived. I will go with `isFilterMatched` for now

**Review by philipp-classen — APPROVED** [review]

**Pull request merged**